src/platform/Darwin/BUILD.gn

```diff
@@ -47,6 +47,8 @@ static_library("Darwin") {
     "ConfigurationManagerImpl.h",
     "ConnectivityManagerImpl.cpp",
     "ConnectivityManagerImpl.h",
+    "DeviceInstanceInfoProviderImpl.cpp",
+    "DeviceInstanceInfoProviderImpl.h",
```

Comment:
This is only instantiated if !CHIP_DISABLE_PLATFORM_KVS, right?  So these should only be compiled if `chip_disable_platform_kvs == false`, I would think....